### DISCUSSION TITLE
GPX recording has gaps despite GPS signal

### DISCUSSION THREAD

**Discussion by HubMiner** [osmandapp/OsmAnd#11816] [General]
### Description
Device is in Adaptive Power mode.  Not directly running any other apps (besides music player).
OsmAnd is added to never sleep apps list.
Problem persists event when the screen is kept on, and the screen shows believable speed and moving position arrow - implying good GPS signal.  Sometimes, however, GPS signal seems lost.  At that time, other GPS-related apps show GPS position, speed just fine.

When recording a hike, after few initial minutes with good behavior, track recording randomly stops for a few minutes, and randomly resumes later.  I tried turning off GPS, restarting the phone - I could not see a pattern,  I tried changing track resolution - continuous, every 1 second - this does not seem to help.

This behavior started a few months ago, before that, this would record a good quality GPX.

### Your Environment
OsmAnd Version: 3.9.10
Android/iOS version: Android 11
Device model: Samsung S10

**Maps used (online or offline):**   
If you have an issue related to offline maps, tell us the exact name of the map file where the issue occurs and its edition date.
Offline maps.

**Comment by gsosna** [comment]
Similar behaviour on Android 11 on Samsung A32. I've noticed that apps that need GPS when not in foreground have an extra option in permissions to allow it to use GPS in background. OSMAND doesn't have that option. I researched further and found ACCESS_BACKGROUND_LOCATION permission should be requested by OSMAND for that, but it's not. Could be related to https://github.com/osmandapp/OsmAnd/discussions/11596

**Reply by gsosna** [reply]
![Screenshot_20210705-093728_Permission controller](https://user-images.githubusercontent.com/7143973/124401125-db022980-dd7b-11eb-93bb-657675c9c231.jpg) ![Screenshot_20210705-093800_Permission controller](https://user-images.githubusercontent.com/7143973/124401128-dc335680-dd7b-11eb-914b-f744b7b2d73b.jpg)

**Reply by sonora** [reply]
The difference between the screenshots is that the "Access all the time permission", a.k.a. ACCESS_**BACKGROUND**_LOCATION, allows an app to access location data without notifying the user about this (Android's definition of "Background").

OsmAnd does not do this: At all times when OsmAnd needs access to the location data, and the app is not visible on the screen, OsmAnd shows a persistent notification in the system notification bar making you aware (the notification about the ongoing navigation and/or the ongoing track recording.) This kind of behavior is in Android terms considered "foreground behavior", and with this transparent behavior OsmAnd can access location data even while the app is not visible on the screen (i.e. after the user taps the device's HOME button) or even after the device screen is turned off.

**Comment by tommycrock** [comment]
I don't get this problem on Galaxy s10e with Android 11. When I am recording a track it gives me a persistent notification, which I think is enough to not need 'all the time' permissions. I believe only Google apps can have all the time permissions.

**Comment by gsosna** [comment]
![Screenshot_20210705-183006_Permission controller](https://user-images.githubusercontent.com/7143973/124430223-03614680-ddc3-11eb-8edd-7555db6e21d0.jpg)

More information on the ACCESS_BACKGROUND_LOCATION permission here: https://developer.android.com/reference/android/Manifest.permission#ACCESS_BACKGROUND_LOCATION
